presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -427,6 +427,20 @@ private QueryStats getQueryStats(Optional<StageInfo> rootStage)
             elapsedTime = nanosSince(createNanos);
         }
 
+        Duration executionTime;
```

Comment:
> Query execution begins when the query is transitioned to planning.

This is a bit of an editorial decision.. it's not super clear to me that execution begins after planning vs after resource-waiting. Intuitively, I think execution should begin when we start actually processing data. I don't feel super strongly about this, but I'm curious to hear your thoughts.

Reply:
In the UIs, when we were presenting execution time, we were showing `elapsed - queued`, which until resource waiting was introduced, meant that execution begins when planning starts.

In my mind there are current 3 top level phases: queued, resource acquisition, and execution. Then I would divide execution into: planning, startup (until full scheduled), fully running, finishing (all tasks finished to query finished), and possibly committing (not sure that can be separated).  To me, planning is part of execution, just like analysis, distributed planning, optimizing, and so on.

Reply:
Makes sense.